clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -5294,24 +5294,18 @@ static void handleNoGlobalWorkOffsetAttr(Sema &S, Decl *D,
 
   checkForDuplicateAttribute<SYCLIntelNoGlobalWorkOffsetAttr>(S, D, Attr);
 
-  uint32_t Enabled = 1;
-  if (Attr.getNumArgs()) {
-    const Expr *E = Attr.getArgAsExpr(0);
-    if (!checkUInt32Argument(S, Attr, E, Enabled, 0,
-                             /*StrictlyUnsigned=*/true))
-      return;
-  }
-  if (Enabled > 1)
-    S.Diag(Attr.getLoc(), diag::warn_boolean_attribute_argument_is_not_valid)
-        << Attr;
-
   if (Attr.getKind() == ParsedAttr::AT_SYCLIntelNoGlobalWorkOffset &&
       checkDeprecatedSYCLAttributeSpelling(S, Attr))
     S.Diag(Attr.getLoc(), diag::note_spelling_suggestion)
         << "'intel::no_global_work_offset'";
 
-  D->addAttr(::new (S.Context)
-                 SYCLIntelNoGlobalWorkOffsetAttr(S.Context, Attr, Enabled));
+  // If no attribute argument is specified, set to default value '1'.
+  Expr *E = Attr.isArgExpr(0)
+                ? Attr.getArgAsExpr(0)
```

Comment:
In the case where `isArgExpr(0)` returns a nonnull pointer, we need to validate that the expression is either dependent (so we'll check it when instantiating the template) or is an integer constant expression.

Reply:
Inside Sema.h , we have check like this :

template <typename AttrType>
void Sema::addIntelSYCLSingleArgFunctionAttr(Decl *D,
                                             const AttributeCommonInfo &CI,
                                             Expr *E) {
  assert(E && "Attribute must have an argument.");

  if (!E->isInstantiationDependent()) {
    Optional<llvm::APSInt> ArgVal = E->getIntegerConstantExpr(getASTContext());
    if (!ArgVal) {
      Diag(E->getExprLoc(), diag::err_attribute_argument_type)
          << CI.getAttrName() << AANT_ArgumentIntegerConstant
          << E->getSourceRange();
      return;
    }

Reply:
Ah, thank you for pointing that out (GitHub wouldn't show me enough file context to spot that). I agree -- that code covers this case.